src/internal/packager/images/pull.go

```diff
@@ -55,178 +53,118 @@ func (i *ImgConfig) PullAll() (map[name.Tag]v1.Image, error) {
 	}
-	spinner.Success()
+	spinner.Updatef("Preparing image sources and cache for image pulling")
 
-	progress := make(chan v1.Update, 200)
+	var (
+		progress    = make(chan v1.Update, 200)
+		progressBar *message.ProgressBar
+		title       string
+	)
 
 	go func() {
 		_ = tarball.MultiWriteToFile(i.TarballPath, tagToImage, tarball.WithProgress(progress))
 	}()
 
-	var progressBar *message.ProgressBar
-	var title string
-
 	for update := range progress {
 		switch {
 		case update.Error != nil && errors.Is(update.Error, io.EOF):
 			progressBar.Success("Pulling %d images (%s)", len(imageMap), utils.ByteFormat(float64(update.Total), 2))
-			return tagToImage, nil
+			return nil
 		case update.Error != nil && strings.HasPrefix(update.Error.Error(), "archive/tar: missed writing "):
 			// Handle potential image cache corruption with a more helpful error. See L#54 in libexec/src/archive/tar/writer.go
 			message.Warnf("Potential image cache corruption: %s of %v bytes - try clearing cache with \"zarf tools clear-cache\"", update.Error.Error(), update.Total)
-			return nil, fmt.Errorf("failed to write image tarball: %w", update.Error)
+			return fmt.Errorf("failed to write image tarball: %w", update.Error)
 		case update.Error != nil:
-			return nil, fmt.Errorf("failed to write image tarball: %w", update.Error)
+			return fmt.Errorf("failed to write image tarball: %w", update.Error)
 		default:
 			title = fmt.Sprintf("Pulling %d images (%s of %s)", len(imageMap),
 				utils.ByteFormat(float64(update.Complete), 2),
 				utils.ByteFormat(float64(update.Total), 2),
 			)
 			if progressBar == nil {
+				spinner.Success()
 				progressBar = message.NewProgressBar(update.Total, title)
 			}
 			progressBar.Update(update.Complete, title)
 		}
 	}
 
-	return tagToImage, nil
+	return nil
 }
 
-// pullImage returns a v1.Image either by loading a local tarball, the pulling from the local daemon, or the wider internet
-func (i *ImgConfig) pullImage(src string) (v1.Image, error) {
-	// Load image tarballs from the local filesystem
+// PullImage returns a v1.Image either by loading a local tarball or the wider internet.
+func (i *ImgConfig) PullImage(src string, spinner *message.Spinner) (img v1.Image, err error) {
+	// Load image tarballs from the local filesystem.
 	if strings.HasSuffix(src, ".tar") || strings.HasSuffix(src, ".tar.gz") || strings.HasSuffix(src, ".tgz") {
-		message.Debugf("loading image tarball: %s", src)
+		spinner.Updatef("Reading image tarball: %s", src)
 		return crane.Load(src, config.GetCraneOptions(true)...)
 	}
 
-	// Unless disabled, attempt to pull the image from the local daemon
-	if !i.NoLocalImages {
+	// If crane is unable to pull the image, try to load it from the local docker daemon.
+	if _, err := crane.Manifest(src, config.GetCraneOptions(i.Insecure)...); err != nil {
+		message.Debugf("crane unable to pull image %s: %s", src, err)
+		spinner.Updatef("%s not found, trying with docker instead. This may take some time.", src)
```

Comment:
Can we make it clearer that we are checking online here?
```suggestion
		spinner.Updatef("%s not found online, trying with docker instead. This may take some time.", src)
```

Reply:
yeah but this also could just be some funky auth thing that we are falling back to, so it could still be online. Any suggestion?

Reply:
Maybe just `Falling back to docker for %s. This may take some time.` Since we can't accurately state the reason?